server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -1381,7 +1377,6 @@ public Source getSource() {
         private void write(final StreamOutput out) throws IOException {
             final int format = SERIALIZATION_FORMAT;
```

Comment:
Same here. I think we need to use the old format for an old wire version. This should be:

```java
final int format = out.getVersion().onOrAfter(Version.V_8_0_0) ? SERIALIZATION_FORMAT : FORMAT_NO_VERSION_TYPE;
```